File: convertor.c
/*
Criado por relue271
Convertor de bases
*/


#include <stdio.h>
#include <string.h>



void main(void){
    int decimal;
    printf("\nDigite um numero inteiro: ");
    scanf("%d", &decimal);







    //converte o decimal para binario
    int aux, i, resto;
    aux = decimal;
    i=0;
    char binario[1024] = "";
    do{
        resto = aux % 2;
        aux = aux / 2;
        binario[i] = resto+'0';
        i++;
    }while(aux > 0);
    binario[i] = '\0';
    printf("\nB: ");
    for (i=strlen(binario); i>=0; i--) {
            if ((i % 4) == 0) { printf(" ");}
            printf("%c", binario[i]);
    }
    printf(" .");






    //converte o decimal para hexadecimal
    aux = decimal;
    i=0;
    char hexa[256] = "";
    do {
        resto = aux % 16;
        aux = aux / 16;
        switch (resto) {
        case 10:
            hexa[i] = 'A';
            break;
        case 11:
            hexa[i] = 'B';
            break;
        case 12:
            hexa[i] = 'C';
            break;
        case 13:
            hexa[i] = 'D';
            break;
        case 14:
            hexa[i] = 'E';
            break;
        case 15:
            hexa[i] = 'F';
            break;
        default:
            hexa[i] = resto+'0';
        }
        i++;
    }while(aux > 0);
    hexa[i] = '\0';
    printf("\nH: ");
    for (i=strlen(hexa); i>=0; i--) { printf("%c", hexa[i]);}
    printf(" .");






    //converte decimal para octal
    aux = decimal;
    i=0;
    char octal[256] = "";
    do{
        resto = aux % 8;
        aux = aux / 8;
        octal[i] = resto+'0';
        i++;
    }while(aux > 0);
    octal[i] = '\0';
    printf("\nO: ");
    for (i=strlen(octal); i>=0; i--) { printf("%c", octal[i]); }
    printf(" .\n\n");
}

File: convertor.go
// Criado por relue271
// Convertor de bases

package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Printf("-")
	convertor()
	fmt.Printf("hello")
}

func convertor() {

	var decimal int
	fmt.Print("Decimal: ")
	fmt.Scanf("%d", &decimal)
	//fmt.Print(decimal)

  
  
  
  
  
	//converte o decimal para binario
	var aux int
	var resto int
	aux = decimal
	resto = decimal

	var binario string
	binario = ""
	for {
		resto = aux % 2
		aux = aux / 2
		binario = strconv.Itoa(resto) + binario //conversao int para char
		if aux <= 0 {
			break
		}
	}
	fmt.Print("B: ")
	fmt.Println(binario)

  
  
  
  
  
	//converte o decimal para hexadecimal
	aux = decimal
	resto = decimal
	var hexa string
	hexa = ""
	for {
		resto = aux % 16
		aux = aux / 16
		switch resto {
		case 10:
			hexa = "A" + hexa
		case 11:
			hexa = "B" + hexa
		case 12:
			hexa = "C" + hexa
		case 13:
			hexa = "D" + hexa
		case 14:
			hexa = "E" + hexa
		case 15:
			hexa = "F" + hexa
		default:
			hexa = strconv.Itoa(resto) + hexa //conversao int para char
		}
		if aux <= 0 {
			break
		}
	}
	fmt.Print("H: ")
	fmt.Println(hexa)

  
  
  
  
  
	//converte decimal para octal
	aux = decimal
	resto = decimal
	var octal string
	octal = ""
	for {
		resto = aux % 8
		aux = aux / 8

		octal = strconv.Itoa(resto) + octal
		if aux <= 0 {
			break
		}
	}
	fmt.Print("O: ")
	fmt.Println(octal)

	return
}
